File: 2_back-end/bytebank/Conta.js
export default class Conta {
  static numeroDeContas = 0;
  _numero;

  constructor(saldoInicial, agencia, cliente) {
    Conta.numeroDeContas += 1
    this._numero = Conta.numeroDeContas;
    this._agencia = agencia;
    this._saldo = saldoInicial;
    this._cliente = cliente;
  };

  get saldo(){
    return this._saldo
  };
  
  set cliente(novoValor){
    if (novoValor instanceof Cliente) {
      this._cliente = novoValor;
    };
  };
  
  get cliente(){
    return this._cliente;
  };
  
  sacar(valor){
    let taxa = 1;
    return this._sacar(valor, taxa);
  };

  _sacar(valor, taxa) {
     const valorSacado = taxa * valor;

    if (this._saldo >= valorSacado){
      this._saldo -= valorSacado;
      return valorSacado;
    };

    return 0;
  }

  depositar(valor) {
    // valor <= 0 ? console.log(`Não é possível depositar ${valor}: valor negativo`) : this._saldo += valor;

    if (valor >= 0){
      this._saldo += valor;
      console.log(`Depósito realizado com sucesso no valor de R$ ${valor},00`);
    } else {
      console.log(`Não é possível depositar R$ ${valor},00 : valor negativo`);
    }
  };

  transferir(valor, conta) {
    const valorSacado = this.sacar(valor);
    conta.depositar(valorSacado);
  };
}

File: 2_back-end/java/bytebank/src/Conta.java
public class Conta {
	private double saldo;
	private int agencia;
	private int numero;
	private Cliente titular;
	private static int totalContas;
	
	public Conta(int agencia, int numero) {
		if (agencia > 0 || numero > 0) {
			totalContas++;
			this.agencia = agencia;
			this.numero = numero;		
			System.out.println("Conta criada com sucesso, agência: " + 
					this.agencia + ", conta: " + this.numero + ", saldo: " + this.saldo);
			System.out.println("Foram criadas " + totalContas + " contas no total.");
		}
	}
	
	public void deposita(double valor) {
		this.saldo += valor;
	}
	
	public boolean saca(double valor) {
		if (this.saldo >= valor) {
			this.saldo -= valor;
			return true;
		} else {
			return false;
		}
	}
		
	public boolean transfere(double valor, Conta destino) {
		saca(valor);
		destino.deposita(valor);
		return true;
	}
	
	public double getSaldo() {
		return this.saldo;
	}
	
	public int getAgencia() {
		return this.agencia;
	}
	
	/* Uma conta é criada já com um número de agência e conta (no constructor), não podendo ser alterarada após criada
	
	public void setAgencia(int agencia) {
		if (agencia > 0) this.agencia = agencia;
	}
	*/
	
	public int getNumero() {
		return this.numero;
	}
	
	/* Uma conta é criada já com um número de agência e conta (no constructor), não podendo ser alterarada após criada
	
	public void setNumero(int numero) {
		if (numero > 0) this.numero = numero;
	}
	*/
	
	public Cliente getTitular() {
		return this.titular;
	}
	
	public void setTitular(Cliente titular) {
		this.titular = titular;
	}
}